network/src/main/java/JarProtocolHandler.java

```diff
@@ -0,0 +1,79 @@
+ *
+ *  Redistribution and use in source and/or binary forms, with or without
+ *  modification, must retain the above copyright notice and the following
+ *  disclaimer.
+ *
+ *  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ *  "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ *  LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+ *  A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+ *  OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ *  SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ *  LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ *  DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ *  THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ *  (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ *  OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+import com.teamdev.jxbrowser.chromium.*;
+import com.teamdev.jxbrowser.chromium.swing.BrowserView;
+
+import javax.swing.*;
+import java.awt.*;
+import java.io.DataInputStream;
+import java.io.InputStream;
+import java.net.URL;
+
+public class JarProtocolHandler {
+    public static void main(String[] args) {
+        final Browser browser = new Browser();
+        BrowserView view = new BrowserView(browser);
+
+        JFrame frame = new JFrame();
+        frame.setDefaultCloseOperation(WindowConstants.EXIT_ON_CLOSE);
+        frame.add(view, BorderLayout.CENTER);
+        frame.setSize(700, 500);
+        frame.setLocationRelativeTo(null);
+        frame.setVisible(true);
+
+        BrowserContext browserContext = browser.getContext();
+        ProtocolService protocolService = browserContext.getProtocolService();
+        protocolService.setProtocolHandler("jar", new ProtocolHandler() {
+            @Override
+            public URLResponse onRequest(URLRequest request) {
+                try {
+                    URLResponse response = new URLResponse();
+                    URL path = new URL(request.getURL());
+                    InputStream inputStream = path.openStream();
+                    DataInputStream stream = new DataInputStream(inputStream);
+                    byte[] data = new byte[stream.available()];
+                    stream.readFully(data);
+                    response.setData(data);
+                    String mimeType = getMimeType(path.toString());
+                    response.getHeaders().setHeader("Content-Type", mimeType);
+                    return response;
+                } catch (Exception ignored) {}
+                return null;
+            }
+        });
+
+        // Assume that we need to load a resource related to this class in the JAR file
+        browser.loadURL(ProtocolHandler.class.getResource("index.html").toString());
+    }
+
+    private static String getMimeType(String path) {
```

Comment:
@vladimir-ikryanov, why don't we use [`MimeType` ](https://docs.oracle.com/javase/7/docs/api/javax/activation/MimeType.html)?